CONTRIBUTING.md

```diff
@@ -0,0 +1,21 @@
+## Contributing
+Contributions of more lockfree data structures are appreciated! Please create issues/PRs to this repo.
```

Comment:
Are we sure we want to restrict this repo to be lock-free only and have another, concurrent one? It does not seem like a natural boundary to me, and I struggle to see how that benefits community or us. Here's a bunch of reasons:
* Many structures are partly lock-free. Are we going to have them duplicated in both (since only concurrent repo can have the extra feature/performance)? 
* We're going to spend a lot of time arguing over lock-freedom. It is not trivial.
    * Any SPMC/SPSC/MPSC structure we have are either incorrect or not lock-free formally. 
    * We had a test run with MPMC queue. It took many days and multiple people voicing opinions to decide even though it came from a paper. That's because it used a way of showing lock-freedom, which is common in the literature, but does not flow from wikipedia definition. 
* Testing and benchmarking for lock-free and "just" concurrent is the same (e.g. obstruction-free still requires lock-free tooling like dscheck). We'll have to dupe them in both repos, and they makes the setups prone to drifting apart. This is bad, as we really need a robust way of comparing fine-grained and locked structures. 
* It's not typical and it's going to be extra confusing for users. 

Is there a good reason to accept this split? 

I imagine there's some target demographics for lock-freedom (or, in particular, wait-freedom) like real-time devs, but we do have a big non-lock-free gc that can kick in at any safe-point. And then, such a demographic can be served by making a table of provided property in the README (lockfree, lockfree if precondidtion, lockfree if at most one consumer, ..).